sdk/Lusid.Sdk/Model/WeightedInstrument.cs

```diff
@@ -1,65 +1,68 @@
  * Contact: info@finbourne.com
  * Generated by: https://github.com/openapitools/openapi-generator.git
  */
 
+
 using System;
-using System.Linq;
-using System.IO;
-using System.Text;
 using System.Collections;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using System.Linq;
+using System.IO;
 using System.Runtime.Serialization;
+using System.Text;
+using System.Text.RegularExpressions;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+using System.ComponentModel.DataAnnotations;
 using OpenAPIDateConverter = Lusid.Sdk.Client.OpenAPIDateConverter;
 
 namespace Lusid.Sdk.Model
 {
     /// <summary>
     /// Specification for a holding or quantity of (weight for) an instrument on a given date.
     /// </summary>
-    [DataContract]
-    public partial class WeightedInstrument :  IEquatable<WeightedInstrument>
+    [DataContract(Name = "WeightedInstrument")]
+    public partial class WeightedInstrument : IEquatable<WeightedInstrument>
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="WeightedInstrument" /> class.
         /// </summary>
         /// <param name="quantity">The quantity of the instrument that is owned..</param>
         /// <param name="holdingIdentifier">Identifier for the instrument.  For a single, unique trade or transaction this can be thought of as equivalent to the transaction identifier, or  a composite of the sub-holding keys for a regular sub-holding. When there are multiple transactions sharing the same underlying instrument  such as purchase of shares on multiple dates where tax implications are different this would not be the case.    In an inlined aggregation request if this is wanted to identify a line item, it can be specified in the set of aggregation keys given on the aggregation  request that accompanies the set of weighted instruments..</param>
         /// <param name="instrument">instrument.</param>
-        public WeightedInstrument(decimal? quantity = default(decimal?), string holdingIdentifier = default(string), LusidInstrument instrument = default(LusidInstrument))
+        public WeightedInstrument(decimal quantity = default(decimal), string holdingIdentifier = default(string), LusidInstrument instrument = default(LusidInstrument))
         {
-            this.HoldingIdentifier = holdingIdentifier;
             this.Quantity = quantity;
             this.HoldingIdentifier = holdingIdentifier;
             this.Instrument = instrument;
         }
-        
+
         /// <summary>
         /// The quantity of the instrument that is owned.
         /// </summary>
         /// <value>The quantity of the instrument that is owned.</value>
-        [DataMember(Name="quantity", EmitDefaultValue=false)]
-        public decimal? Quantity { get; set; }
+        [DataMember(Name = "quantity", EmitDefaultValue = true)]
+        public decimal Quantity { get; set; }
```

Comment:
What is the signifiance of EmitDefaultValue being changed to `true`? - https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datamemberattribute.emitdefaultvalue?view=netcore-3.1

It looks to me that all required parameters now have this set to true and for all optional parameters it remains set to false. Looking at the options (https://openapi-generator.tech/docs/generators/csharp-netcore/) I think we can only affect the behaviour of the optional parameters so this change should be fine.

Reply:
I made this change because there is a change in how the JSON serialisation code works. It's in this function:

```
        private bool ShouldWriteProperty(object? memberValue, JsonObjectContract? containerContract, JsonProperty property)
        {
            if (memberValue == null && ResolvedNullValueHandling(containerContract, property) == NullValueHandling.Ignore)
            {
                return false;
            }
            if (HasFlag(property.DefaultValueHandling.GetValueOrDefault(Serializer._defaultValueHandling), DefaultValueHandling.Ignore)
                && MiscellaneousUtils.ValueEquals(memberValue, property.GetResolvedDefaultValue()))
            {
                return false;
            }
            return true;
        }
```

So what happens is that if one of your fields uses the default value for that field (e.g. `rateOrSpread: 0.00m`) then that field won't be included in the request due to it having the default value...unless you set `EmitDefaultValue = true`. 

It only affects integers and decimals.

Reply:
Makes sense, great!